time/time.go

```diff
@@ -0,0 +1,43 @@
+package time
+
+import (
+	"time"
+)
+
+const (
+	// dateTimeFormat is a IMF-fixdate formatted time https://tools.ietf.org/html/rfc7231.html#section-7.1.1.1
+	dateTimeFormat = "2006-01-02T15:04:05.99Z"
+
+	// httpDateFormat is a date time defined by RFC3339 section 5.6 with no UTC offset.
+	httpDateFormat = "Mon, 02 Jan 2006 15:04:05 GMT"
+)
+
+// FormatDateTime format value as a date-time
+func FormatDateTime(value time.Time) string {
+	return value.Format(dateTimeFormat)
```

Comment:
Minor it might be helpful to reference the RFC, or example format these formats correspond to. Will make it easier to read/use these in the future.

Reply:
Per discussion to ease some other outstanding PRs I've moved the time code to https://github.com/awslabs/smithy-go/pull/53 and addressed the feedback there.